test/nofiles.test.js

```diff
@@ -0,0 +1,90 @@
+const stylelint = require('stylelint')
+const path = require('path')
+
+const code = `import styled from 'styled-components';
+const StyledTable = styled(StyledTableBase)\`
+color: red;
+\``
+const processor = path.join(__dirname, '../lib/index.js')
+const rules = {
+  indentation: 2
+}
+
+describe('no files', () => {
+  let codeFilename
+  let data
+
+  // NOTE beforeEach() runs _after_ the beforeAll() hooks of the describe() blocks
+  beforeEach(done => {
+    stylelint
+      .lint({
+        code,
+        codeFilename,
+        syntax: 'scss',
+        config: {
+          processors: [processor],
+          rules
+        }
+      })
+      .then(result => {
+        data = result
+        done()
+      })
+      .catch(err => {
+        // eslint-disable-next-line
+        console.log(err)
+        data = err
+        done()
+      })
+  })
+
+  describe('no codeFilename', () => {
+    beforeAll(() => {})
+
+    it('should have one result', () => {
+      expect(data.results.length).toEqual(1)
+    })
+
+    it('should use the right file', () => {
+      expect(data.results[0].source).toEqual('<input css 2>')
+    })
+
+    it('should have errored', () => {
+      expect(data.errored).toEqual(true)
+    })
+
+    it('should have one warnings', () => {
```

Comment:
Very picky, but I guess there shouldn't be an s on the warning, I might have let it go if I hadn't asked for another change though.

Reply:
The more picky, the better. Thanks for pointing out!